src/server/modes/charts/plugins/datalens/preparers/metric/index.ts

```diff
@@ -1,92 +1,37 @@
-import {dateTime} from '@gravity-ui/date-utils';
+import {Feature, isEnabledServerFeature} from '../../../../../../../shared';
+import {registry} from '../../../../../../registry';
+import {findIndexInOrder} from '../../utils/misc-helpers';
+import {PrepareFunctionArgs} from '../types';
 
-import {
-    CommonNumberFormattingOptions,
-    MINIMUM_FRACTION_DIGITS,
-    NumberFormatType,
-    NumberFormatUnit,
-    getFakeTitleOrTitle,
-    isDateField,
-} from '../../../../../../../shared';
-import {findIndexInOrder, isFloatDataType, isNumericalDataType} from '../../utils/misc-helpers';
-
-import {PrepareFunctionArgs} from './../types';
-
-type MetricCurrent = {
-    value: string | number | null;
-    precision?: number;
-    format?: NumberFormatType;
-    postfix?: string;
-    prefix?: string;
-    showRankDelimiter?: boolean;
-    unit?: NumberFormatUnit;
-};
-
-type MetricConfig = {
-    content: {
-        current: MetricCurrent;
-    };
-    title: string;
-    size: string;
-    color: string;
-};
+import {prepareBasicMetricVariant} from './variants/basic';
+import {prepareMarkupMetricVariant} from './variants/markup';
 
 function prepareMetric({placeholders, resultData, shared, idToTitle}: PrepareFunctionArgs) {
     const {data, order} = resultData;
 
     const measure = placeholders[0].items[0];
 
+    if (typeof measure === 'undefined') {
+        return {};
```

Comment:
Is this a valid return result? What will be displayed in this case?

Reply:
Yes. Chartkit will display "No data"